lib/http-bridge/test/integration/Cardano/Wallet/HttpBridge/NetworkSpec.hs

```diff
@@ -144,6 +155,27 @@ spec = do
                     "Transaction failed verification: transaction has more \
                     \witnesses than inputs"
             runExceptT (postTx bridge signed) `shouldReturn` err
+
+    describe "waitForConnection" $ do
+        it "times out after a short while" $ do
+            nw <- newNetworkLayer
+            waitForConnection nw `shouldThrow` \case
+                ErrNetworkTipNetworkUnreachable _ -> True
```

Comment:
This waits for the full timeout (20 s) in CI, which is not great, but I didn't find the alternatives obvious

- Don't test this
- Provide the timeout as a parameter

Well it would be the last one I guess 🤔

Reply:
Added a timeout in seconds.

Reply:
Hmmm.. This is dodgy as we just don't know how long it's going to take in practice. It really depends on the host machine... I wouldn't have a test for that, or clearly not a test that is time-based. Unless I am misunderstanding something. If the bridge isn't available here and we are sure that this will fail, what's the need for a delay?